xonsh/prompt/vc.py

```diff
@@ -101,27 +102,26 @@ def get_hg_branch(root=None):
         return None
     if env.get("VC_HG_SHOW_BRANCH"):
         # get branch name
-        branch_path = os.path.sep.join([root, ".hg", "branch"])
-        if os.path.exists(branch_path):
+        branch_path = Path(root) / ".hg" / "branch"
+        if branch_path.exists():
             with open(branch_path, "r") as branch_file:
-                branch = branch_file.read()
+                branch = branch_file.read().strip()
         else:
             branch = "default"
     else:
         branch = ""
-    # add bookmark, if we can
-    bookmark_path = os.path.sep.join([root, ".hg", "bookmarks.current"])
-    if os.path.exists(bookmark_path):
-        with open(bookmark_path, "r") as bookmark_file:
-            active_bookmark = bookmark_file.read()
-        if env.get("VC_HG_SHOW_BRANCH") is True:
-            branch = "{0}, {1}".format(
-                *(b.strip(os.linesep) for b in (branch, active_bookmark))
-            )
-        else:
-            branch = active_bookmark.strip(os.linesep)
-    else:
-        branch = branch.strip(os.linesep)
+    # add activated bookmark and topic
+    for filename in ["bookmarks.current", "topic"]:
+        feature_branch_path = Path(root) / ".hg" / filename
+        if feature_branch_path.exists():
+            with open(feature_branch_path) as file:
+                feature_branch = file.read().strip()
+            if feature_branch:
+                if branch:
+                    branch = f"{branch}, {feature_branch}"
```

Comment:
Since a topic belongs to a branch, it is possible to use a slash instead of a comma like
````python
branch = f"{branch}/{feature_branch}"
````
This is the notation from heptapod, e.g. . https://foss.heptapod.net/mercurial/hg-git where you can select the different (topic-) branches.

Reply:
Yes! But we should keep the comma notation for the bookmarks...